optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/solution/descriptor/ClassBrowser.java

```diff
@@ -0,0 +1,55 @@
+package org.optaplanner.core.impl.domain.solution.descriptor;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.*;
+
+class ClassBrowser {
```

Comment:
All classes are public, to encourage reuse of classes with optaplanner itself, across packages.

Note: not all classes are considered "module public" - but until JDK 9 we can't enforce that.